### ISSUE TITLE
lineSplit does not split a lineString if splitter is very close to a vertex point

### ISSUE THREAD

**Issue by grundid** [Turfjs/turf#852]
The following test returns a feature collection with only one element.
I would expect two elements.

```javascript
const lineString = turf.lineString([[9.2202022, 49.1438226], [9.2199531, 49.1439048], [9.2196177, 49.1440264]]);
const resultFc = turf.lineSplit(lineString, turf.point([9.219953175634146, 49.14390480825327]));
t.equals(resultFc.features.length, 2);
```

When I reduce the precision of the splitter the result is correct:
```javascript
const lineString = turf.lineString([[9.2202022, 49.1438226], [9.2199531, 49.1439048], [9.2196177, 49.1440264]]);
const resultFc = turf.lineSplit(lineString, turf.point([9.2199531, 49.1439048]));
t.equals(resultFc.features.length, 2);
```

**Comment by stebogit** [comment]
@grundid it seems to be a matter of decimals in the splitter (`point`) coordinates:
```js
const lineString = turf.lineString([[9.2202022, 49.1438226], [9.2199531, 49.1439048], [9.2196177, 49.1440264]]);
const resultFc = turf.lineSplit(lineString, turf.point([9.219953, 49.143905])); // coordinates truncated to 6 decimals
//={ "type": "FeatureCollection", "features": [ { "type": "Feature", "properties": {}, "geometry": { "type": "LineString", "coordinates": [ [ 9.2202022, 49.1438226 ], [ 9.2199531, 49.1439048 ], [ 9.219953, 49.143905 ] ] } }, { "type": "Feature", "properties": {}, "geometry": { "type": "LineString", "coordinates": [ [ 9.219953, 49.143905 ], [ 9.2196177, 49.1440264 ] ] } } ] }
```

Turf follows GeoJSON [recommendations](https://tools.ietf.org/html/rfc7946#section-11.2) by using 6 decimals to define coordinates.

@DenisCarriere we might want to `truncate` the coords of the splitter in`@turf/line-split` to avoid this issue?

**Comment by grundid** [comment]
If Turf follows the GeoJSON recommendations then maybe turf.point should not create a GeoJSON object with to many decimals. ;)

**Comment by DenisCarriere** [comment]
> then maybe turf.point should not create a GeoJSON object with to many decimals
Might not be a good idea to add `truncate` to `turf.point` since that would add unnecessary overhead if it's not required.

>  we might want to truncate the coords of the splitter in@turf/line-split to avoid this issue?

👍 Agreed, for this scenario we might want to introduce `truncate` for the splitter, we can add this exact example to the `test.js`.

@grundid Good catch!

**Comment by DenisCarriere** [comment]
Fixed in the next minor release.

**Comment by grundid** [comment]
After the latest release lineSplit became somehow quite useless for my cases:

Case 1:
```javascript
  const lineString = turf.lineString([[9.2202022, 49.1438226], [9.2199531, 49.1439048], [9.2196177, 49.1440264]]);
  const resultFc = turf.lineSplit(lineString, turf.point([9.2202022, 49.1438226]));
```
Returns two lines. One very short:
line1: [[9.2202022,49.1438226],[9.220202,49.143823]]
line2: [[9.220202,49.143823],[9.2199531,49.1439048],[9.2196177,49.1440264]]
Expected result would be to not cut at all.

Case 2:
```javascript
  const lineString = turf.lineString([[9.2185956, 49.1450893], [9.2181343, 49.1443734], [9.2192501, 49.1438962]]);
  const resultFc = turf.lineSplit(lineString, turf.point([9.2181343, 49.1443734]));
```

Here the cut does not work. lineSplit returns one feature which is the original line.

**Comment by DenisCarriere** [comment]
@grundid Thanks for reporting this, could you add these test cases to the [`turf-line-split/test.js`](https://github.com/Turfjs/turf/blob/master/packages/turf-line-split/test.js#L36-L44) with your expected results, the PR should fail.

**Comment by stebogit** [comment]
Bug still somewhere... 🐛 
See PR #892

**Comment by DenisCarriere** [comment]
Those pesky 🐛 🐞 !

**Comment by DenisCarriere** [comment]
This is fixed in the latest `@turf/line-split@4.6.1`

**Comment by IshanthaKamal** [comment]
Scenario 1 

![untitled document](https://user-images.githubusercontent.com/18345069/38294082-60078134-3807-11e8-95c8-ce237287ca94.png)

Expected Result

![untitled document 3](https://user-images.githubusercontent.com/18345069/38294102-6d5cabe8-3807-11e8-9cd2-1be6869de88a.png)

Scenario 2 

![untitled document 2](https://user-images.githubusercontent.com/18345069/38294117-77c9db14-3807-11e8-8b7f-bd19e10b5bb9.png)

I was trying to get split parts of 2 lines and get the merge.What I did was get line split in line1 vs line2 and again line2 vs line1. Issue is line splitting not working properly when splitting is close to vertex point as in above images.I noticed that line split is not giving proper splitter parts.In last scenario line splitting is giving only 2 splitter parts of line A instead of 3.
